tsfile/src/main/java/org/apache/iotdb/tsfile/write/writer/TsFileIOWriter.java

```diff
@@ -65,14 +66,19 @@
   }
 
   protected TsFileOutput out;
-  protected List<Pair<String, List<ChunkMetadata>>> chunkGroupInfoList = new ArrayList<>();
   protected boolean canWrite = true;
   protected int totalChunkNum = 0;
   protected int invalidChunkNum;
   protected File file;
-  protected List<ChunkMetadata> chunkMetadataList = new ArrayList<>();
-  protected Map<Path, List<ChunkMetadata>> chunkMetadataListMap = new TreeMap<>();
+
+
+  // current flushed Chunk
   private ChunkMetadata currentChunkMetadata;
+  // current flushed ChunkGroup
+  protected List<ChunkMetadata> chunkMetadataList = new ArrayList<>();
+  // all flushed ChunkGroup,  device -> List<ChunkMetadata>
+  protected List<Pair<String, List<ChunkMetadata>>> chunkGroupMetadataList = new ArrayList<>();
+
```

Comment:
Since Pair<String, List<ChunkMetadata> is similar to previous ChunkGroupMetadata, I'm thinking if it's necessary to add ChunkGroupMetadata back instead of using the pair.

Reply:
Good idea, I have added it